client.cpp

```diff
@@ -1,37 +1,77 @@
-#include <hazelcast/client/hazelcast_client.h>
 #include <string>
+#include "model/City.h"
+#include "model/Country.h"
 
+/**
+ * This is boilerplate application that configures client to connect Hazelcast
+ * Cloud cluster.
+ * <p>
+ * See: <a href="https://docs.hazelcast.com/cloud/cpp-client">https://docs.hazelcast.com/cloud/cpp-client</a>
+ */
+
+void mapExample(hazelcast::client::hazelcast_client client);
+void nonStopMapExample(hazelcast::client::hazelcast_client client);
 int main(int argc, char **argv) {
     hazelcast::client::client_config config;
-    
-    config.set_cluster_name("YOUR_CLUSTER_NAME");
-    config.set_property(hazelcast::client::client_properties::CLOUD_URL_BASE, "YOUR_DISCOVERY_URL");
     config.set_property("hazelcast.client.statistics.enabled", "true");
-
+    config.set_property(hazelcast::client::client_properties::CLOUD_URL_BASE, "YOUR_DISCOVERY_URL");
+    config.set_cluster_name("YOUR_CLUSTER_NAME");
     auto &cloud_configuration = config.get_network_config().get_cloud_config();
     cloud_configuration.enabled = true;
     cloud_configuration.discovery_token = "YOUR_CLUSTER_DISCOVERY_TOKEN";
-
-    auto hazelcastClient = hazelcast::new_client(std::move(config)).get();
-    auto map = hazelcastClient.get_map("map").get();
-    auto check = map->put<std::string, std::string>("key", "value").get();
-    map->clear().get();
+    auto client = hazelcast::new_client(std::move(config)).get();
     std::cout << "Connection Successful!" << std::endl;
-    std::cout << "Now, 'map' will be filled with random entries." << std::endl;
 
-    int iterationCount = 0;
+    mapExample(client);
+
+    //nonStopMapExample(client);
+
+    client.shutdown();
+}
+
+/**
+* This example shows how to work with Hazelcast maps.
+*
+* @param client - a {@link HazelcastInstance} client.
```

Comment:
HazelcastInstance does not exist for C++ client. Also, is the `@link` a valid documentation tag for C++?

Reply:
Should be fine now